### ISSUE TITLE
Missing object references in assay_metadata.proto

### ISSUE THREAD

**Issue by mbaudis** [ga4gh/schemas#683]
In ```assay_metadata.proto```,  ```Experiment```does not contain an attribute to link to the respective bio_metadata objects.

For ```Experiment```, the object description says "An experimental preparation of a sample". This implies referencing the corresponding Biosample through ```biosample_id```.

**Comment by david4096** [comment]
Currently, `experiment` is an attribute of a `ReadGroup` that embeds the `Experiment` document in a read group. Read groups currently have a `bio_sample_id`, which allows an experiment to be implicitly tied to a biosample. This means that although the relationship is not explicit, it can be derived.

I think we could close this issue by stating this information is available. Adding it as a field to the Experiment message would introduce a minor redundancy that would lead to few code changes.

Since the `Experiment` message is given an `id` we should consider moving it to its own endpoint, I believe the intention may have been to provide a direct access pattern. This would allow `readgroups` to remain as simple of a message as possible by removing the embedded `experiment` field and replacing it with `experimentId`. One could then access experiments by filtering by biosample.

**Comment by mbaudis** [comment]
@david4096 I am in support of Experiment as separate end point, referenced from whatever other object through experiment_id. Otherwise ```Analysis``` would be difficult to implement, especially  when considering resources not providing reads (e.g. resources just providing variants or using other technologies).

**Comment by mcourtot** [comment]
I'm a little bit confused by the workflow. Are all `Experiment` resulting in `Reads`? Are all `Analysis` done on `Reads` only?

**Comment by mbaudis** [comment]
@mcourtot Exactly. The current "reads" defined (implied) structure assumes this. And even for ```Experiment```s resulting in reads, most API implementations will query variants (but may need assay metadata, from ```Experiment``` and ```Analysis```).

**Comment by mbaudis** [comment]
... but @david4096's suggestion (```experiment_id``` instead of embedded ```experiment```) provides an easy structural fix.

**Comment by mcourtot** [comment]
I like @david4096's suggestion as well as I think it will provide greater flexibility. 

Side question: how is the link with `Analysis` made?

**Comment by mbaudis** [comment]
Both ```Experiment``` & ```Analysis``` had not been defined in a comprehensible manner. In my reading, ```Experiment``` represents the procedures applied to an aliquot of a ```Biosample``` (e.g. library prep, PCR, hybridization...) and leads to ```Reads``` or other raw data files; while ```Analysis``` is what is applied to those files procedurally & covers software & bioinformatic methods (variant calling, filtering, segmentation ...).

Biosample => 1,n Experiment => Reads => 1,n Analysis => Callset

Callset is a bit odd since it represents the subset of variants found in through particular Biosample ... Analysis chain, but being part of a Variantset from possibly many analyses... This is how it shows in VCF, but counterintuitive since the underlying biological reality is independent of other analyses. 

_IMHO one should get rid of the Variantset || Callset wrapper and express each analysis result as a Variantset of 1 (I understand legacy/performance issues here, but context can be added through other data structures)._

**Comment by david4096** [comment]
I also see the value of an experiments endpoint as it provides a way to track multiple efforts within the same dataset. However, given that `Experiment` is a nested document within `ReadGroup` can we close this issue as the reference isn't missing, just implicit? With your support we could begin the effort to model an `experiments` endpoint in a separate issue @mbaudis .

**Comment by mbaudis** [comment]
@david4096 Yes; `Experiment`s, `Analysis`(es) are objects to tackle, and fortunately not really loaded yet (just disregard what is in there).

**Comment by mbaudis** [comment]
@david4096 @kozbo @dzerbino @mcourtot 

Bump for the Experiment/Analysis discussion.

I can't follow the decision to remove `assay_metadata.proto` & put `Experiment`, `Analysis` into common. There was a lengthy discussion involving @dzerbino and others which ended up with this design; And it has a clear logic to it. Accordingly, `Program` should be in `assay_metadata.proto`, too.

Also, the structures of these objects are still ill-defined, based on a very sketchy design early in the schema development. It would be good to have people involved with experimental design (not only sequencing) and technical ontologies to have a go at these.

**Comment by david4096** [comment]
My loose proposal is to add an API and direct access method for Experiment, that will force us to do the modeling so that the object won't be nested improperly in a Read Group Set. What if I only have variant data? How do I model the experiment performed?

Analysis seems like an object that if applied well could have some simplifying effects as it suggests being able to provide scalar values with some hand-authored metadata. However, in practice I found it difficult to implement properly.
